src/solver/modeler/api/include/antares/solver/modeler/api/linearProblemFiller.h

```diff
@@ -30,9 +32,19 @@ namespace Antares::Solver::Modeler::Api
 class LinearProblemFiller
 {
 public:
-    virtual void addVariables(LinearProblem* problem, LinearProblemData* data) = 0;
-    virtual void addConstraints(LinearProblem* problem, LinearProblemData* data) = 0;
-    virtual void addObjectiveCoefficients(LinearProblem* problem, LinearProblemData* data) = 0;
+    explicit LinearProblemFiller(ILinearProblem& pb, LinearProblemData& data):
+        linearProblem_(pb),
+        LPdata_(data)
+    {
+    }
+
+    virtual void addVariables() = 0;
+    virtual void addConstraints() = 0;
+    virtual void addObjective() = 0;
+
+protected:
+    ILinearProblem& linearProblem_;
+    const LinearProblemData& LPdata_;
```

Comment:
this class should stay an interface without member data;
I understand the need of these members, as an alternative put them in the builder

Reply:
and without ctor of course

Reply:
Could you explain your point of view ?
As it is, it makes sense to me : fillers are the only objects in charge of adding variables and constraints to problems.
And there are drawbacks in not having LP and LP data as fillers data members : 
- we must pass these objects each time we call a filler's method, so the code gets more complicated.
- **builder** gets necessarily new dependencies to class **ILinearProblem** and **LinearProblemData**, so it leads to more dependancies.

@pet-mit asks more or less for the same changes in [this comment](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2391#discussion_r1757090621)

Reply:
This class no longer have data members.